tests/src/AsyncIntegrationTests.cpp

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2019 United States Government as represented by the
+// Administrator of the National Aeronautics and Space Administration.
+// All Rights Reserved.
+
+// TODO (JW): By its nature, this file is designed to be copied. It should
+// be explicitely (un)licensed as public domain or CC0.
+#include <chrono>
+#include <ctime>
+#include <fstream>
+#include <iomanip>
+#include <map>
+#include <sstream>
+#include <vector>
+
+#include "Messages/MessageBus.h"
+#include "Messages/ProgEventMessage.h"
+#include "Messages/ScalarMessage.h"
+#include "ModelBasedEventDrivenPrognoserBuilder.h"
+#include "Test.h"
+
+using namespace PCOE;
+
+namespace AsyncIntegrationTests {
+    const size_t ITERATIONS = 12;
+    
+    // This function is a quick and dirty reader for the example data files.
```

Comment:
I would get rid of most of the comments in this file. Since this is test code, they're not as necessary and it's likely that the code will change over time without matching changes to the comments.